src/test/java/org/springframework/integration/aws/inbound/SnsInboundChannelAdapterTests.java

```diff
@@ -133,7 +133,7 @@ public void testUnsubscribe() throws Exception {
 		this.mockMvc.perform(
 				post("/mySampleTopic")
 						.header("x-amz-sns-message-type", "UnsubscribeConfirmation")
-						.contentType(MediaType.APPLICATION_JSON)
+						.contentType(MediaType.TEXT_PLAIN)
```

Comment:
Yeah, I was wondering should I change those... It sounds like a good idea to leave `MediaType.APPLICATION_JSON`, though, the issue was raised with Amazon in 2011, and who knows how long more it will take before it's fixed